### PR TITLE
iOS: split `EnvoyEngine.h` into multiple header files

### PR DESCRIPTION
**Pull request by jpsim**
Resolves https://github.com/envoyproxy/envoy-mobile/issues/333

Commit Message:
Additional Description:
Risk Level: Low, issues should be caught by CI, no risk to non-Objective-C/Swift targets
Testing:
Docs Changes:
Release Notes:
Platform Specific Features: Only affects Objective-C/Swift targets
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: mobile/library/objective-c/EnvoyEventTracker.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+#import "library/objective-c/EnvoyAliases.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface EnvoyEventTracker : NSObject
+
+@property (nonatomic, copy, nonnull) void (^track)(EnvoyEvent *);
+
+- (instancetype)initWithEventTrackingClosure:(nonnull void (^)(EnvoyEvent *))track;
```
Comment (Augustyniak) [inline mobile/library/objective-c/EnvoyEventTracker.h:11]: nit picking: not sure whether `nonnull` attributes in here and at line #9 are needed but it was probably just copied-over and it is not a problem.

**Review by Augustyniak — APPROVED** [review]
one optional comment

File: mobile/library/objective-c/EnvoyEventTracker.h (hunk 1 above)
Comment (jpsim) [reply mobile/library/objective-c/EnvoyEventTracker.h]: Good eye! Removed.

**Review by Augustyniak — APPROVED** [review]

**Pull request merged**